<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1294</width>
    <height>942</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#centralwidget {
	background: #15242d;	

}


QWidget {
	background: rgb(37, 38, 49);	
	color:white;
}
QGroupBox {
	background: #15242d;
	color:rgb(241, 241, 241);
}

QComboBox{
	background: white;
	color:black;
}

QPushButton {
	background: rgb(0, 24, 36);
	color:rgb(241, 241, 241);
}

</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QGroupBox" name="grpbx_DigitalFilter">
      <property name="title">
       <string>Digital Filter Design</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="PlotWidget" name="plot_magResponse" native="true">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="PlotWidget" name="plot_phaseResponse" native="true">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="0">
        <widget class="QWidget" name="wgt_unitCircle" native="true">
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="1" column="4">
           <widget class="PlotWidget" name="plot_unitCircle" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>400</width>
              <height>400</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>400</width>
              <height>400</height>
             </size>
            </property>
            <property name="sizeIncrement">
             <size>
              <width>1</width>
              <height>1</height>
             </size>
            </property>
            <property name="baseSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="btn_Undo">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Undo</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QPushButton" name="btn_Redo">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Redo</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QWidget" name="wgt_buttons" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="leftMargin">
              <number>10</number>
             </property>
             <item>
              <widget class="QPushButton" name="btn_addPoles">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Add Pole</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_addZeros">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Add Zero</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_Swapping">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Swapping</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_removePoles">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Remove Poles</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_RemoveZeros">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>Remove Zeros</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_removeAll">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Remove All</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_Remove">
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Remove</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">btngrp_zerosPoles</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="pair_mode_toggle">
               <property name="text">
                <string>Pair Mode</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="mouse_en">
               <property name="text">
                <string>Mouse Enable</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="btn_Import_Zero_Pole">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Import</string>
            </property>
            <property name="shortcut">
             <string>Ctrl+O</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QPushButton" name="btn_Export_Zero_Pole">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Export</string>
            </property>
            <property name="shortcut">
             <string>Ctrl+O</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="grpbx_AllPassFilter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>All Pass Filter Design</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>All Pass Filters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTableWidget" name="table_coeff">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::DoubleClicked</set>
            </property>
            <property name="alternatingRowColors">
             <bool>false</bool>
            </property>
            <property name="showGrid">
             <bool>true</bool>
            </property>
            <property name="cornerButtonEnabled">
             <bool>false</bool>
            </property>
            <property name="columnCount">
             <number>1</number>
            </property>
            <attribute name="horizontalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <column/>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="PlotWidget" name="plot_allPass" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>70</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>300</width>
           <height>16777215</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QWidget" name="wgt_coefficient" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>70</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>300</width>
           <height>70</height>
          </size>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="all_pass_enable">
            <property name="text">
             <string>All Pass</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="btn_addCoeff">
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Coefficient</string>
            </property>
            <property name="buddy">
             <cstring>wgt_coefficient</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="btn_removeCoeff">
            <property name="text">
             <string>Remove</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QComboBox" name="comboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background:white;
color:black;
</string>
            </property>
            <property name="editable">
             <bool>true</bool>
            </property>
            <item>
             <property name="text">
              <string>-1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>-0.9</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>-0.495</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0.495</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0.9</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="grpbx_RealtimeFiltering">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>300</height>
       </size>
      </property>
      <property name="title">
       <string>Realtime Filtering</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="4">
        <widget class="QPushButton" name="btnClr">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="PlotWidget" name="plot_realtimeFilter" native="true"/>
       </item>
       <item row="0" column="4">
        <widget class="PlotWidget" name="plot_mouseInput" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>200</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btn_openFile">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Choose File</string>
         </property>
         <property name="shortcut">
          <string>Ctrl+O</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="PlotWidget" name="plot_realtimeInput" native="true"/>
       </item>
       <item row="1" column="3">
        <widget class="QSlider" name="speed_slider">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="lbl_speed">
         <property name="text">
          <string>Speed: 1</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buddy">
          <cstring>speed_slider</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="btn_play">
         <property name="text">
          <string>Play</string>
         </property>
         <property name="shortcut">
          <string>Space</string>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionImport">
   <property name="text">
    <string>Import Zeros/Poles</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>Export Zeros/Poles</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="btngrp_zerosPoles"/>
 </buttongroups>
</ui>
